Mail/Views/New Message/Header Cells/ComposeMessageSenderMenu.swift

```diff
@@ -58,23 +59,23 @@ struct ComposeMessageSenderMenu: View {
                     Text(type.title)
                         .textStyle(.bodySecondary)
 
-                    if let currentSignature {
-                        Menu {
-                            ForEach(signatures) { signature in
-                                SenderMenuCell(currentSignature: $currentSignature, signature: signature)
-                            }
-                        } label: {
-                            Text(currentSignature, format: .signature(style: canSelectSignature ? .long : .short))
-                                .textStyle(.body)
-                                .lineLimit(1)
-                                .frame(maxWidth: .infinity, alignment: .leading)
+                    Menu {
+                        SenderMenuCell(currentSignature: $currentSignature, signature: nil)
+                        ForEach(signatures) { signature in
+                            SenderMenuCell(currentSignature: $currentSignature, signature: signature)
+                        }
+                    } label: {
+                        Text(currentSignature?.formatted(style: canSelectSignature ? .long : .short) ?? mailboxManager.mailbox
```

Comment:
Can you transform the Text content into a variable so it would be easier to read?